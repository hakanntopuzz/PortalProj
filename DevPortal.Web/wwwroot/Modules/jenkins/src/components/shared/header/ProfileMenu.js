import React, { Fragment } from "react";
import { UncontrolledDropdown, DropdownToggle, DropdownMenu } from "reactstrap";
import useFetch from "../../../hooks/useFetch";
import { GET_USER_IDENTITY_URL } from "../../../config/urls";

const ProfileMenu = ({ menu }) => {
  let { error, data } = useFetch(GET_USER_IDENTITY_URL, null);
  const profileMenuItems = menu.find((x) => x.group === 2);

  return (
    <Fragment>
      {!error && data && (
        <UncontrolledDropdown
          className="nav-item-profile float-right"
          data-testid="profile-menu">
          <DropdownToggle nav caret data-testid="profileName">
            {data}
          </DropdownToggle>
          {profileMenuItems.children.length > 0 && (
            <DropdownMenu>
              {profileMenuItems.children.map((child, childIndex) => (
                <a
                  key={childIndex}
                  href={`${child.link}`}
                  className="dropdown-item">
                  <i className={`fa ${child.icon}`} aria-hidden="true"></i>
                  {child.name}
                </a>
              ))}
            </DropdownMenu>
          )}
        </UncontrolledDropdown>
      )}
    </Fragment>
  );
};

export default ProfileMenu;
