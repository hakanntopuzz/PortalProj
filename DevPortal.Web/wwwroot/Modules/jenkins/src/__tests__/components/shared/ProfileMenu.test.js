import React from "react";
import { cleanup } from "@testing-library/react";
import ShallowRenderer from "react-test-renderer/shallow";
import ProfileMenu from "../../../components/shared/header/ProfileMenu";

describe("Profile Menu", () => {
  afterEach(cleanup);

  test("renders without crashing", () => {
    const menu = [
      {
        name: "Hesap",
        link: null,
        icon: null,
        group: 2,
        children: [
          {
            name: "Hesap Bilgileri",
            link: "/account/profile",
            icon: "person_outline",
            group: 2,
            children: [],
            hasChildren: false,
          },
          {
            name: "Parola",
            link: "/account/password",
            icon: "lock_outline",
            group: 2,
            children: [],
            hasChildren: false,
          },
          {
            name: "Çıkış Yap",
            link: "/account/logout",
            icon: "exit_to_app",
            group: 2,
            children: [],
            hasChildren: false,
          },
        ],
        hasChildren: true,
      },
    ];
    
    const renderer = new ShallowRenderer();
    renderer.render(<ProfileMenu menu={menu} />);

    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
