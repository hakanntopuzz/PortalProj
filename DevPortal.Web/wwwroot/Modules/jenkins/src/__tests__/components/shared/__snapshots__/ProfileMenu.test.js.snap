// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Menu renders without crashing 1`] = `<React.Fragment />`;
